src/code/PublishPSResource.cs

```diff
@@ -329,27 +353,41 @@ protected override void EndProcessing()
                 }
                 else
                 {
-                    // Create subdirectory structure in temp folder
-                    foreach (string dir in System.IO.Directory.GetDirectories(_path, "*", System.IO.SearchOption.AllDirectories))
+                    try
                     {
-                        var dirName = dir.Substring(_path.Length).Trim(_PathSeparators);
-                        System.IO.Directory.CreateDirectory(System.IO.Path.Combine(outputDir, dirName));
-                    }
+                        // If path is pointing to a file, get the parent directory, otherwise assumption is that path is pointing to the root directory
+                        string rootModuleDir = _path.EndsWith(PSDataFileExt) ? new FileInfo(_path).Directory.FullName : _path;

```

Comment:
We have checked the `_path` file extension previously.
```suggestion
                        string rootModuleDir = _path.EndsWith(PSDataFileExt, StringComparer.OriginalIgnoreCase) ? new FileInfo(_path).Directory.FullName : _path;

```